traits/has_traits.py

```diff
@@ -456,341 +466,322 @@ def remove_listener ( cls, listener, class_name = '' ):
-                        setter = HasTraits._set_traits_cache
-                    elif getattr( getter, 'flushable', False ):
-                        setter = HasTraits._flush_traits_cache
-                validate = _property_method( class_dict, '_validate_' + name )
-                if validate is None:
-                    validate = value.validate
-
-                value = Property( getter, setter, validate, True,
-                                  value.handler, **value.metadata )
-            if rc:
-                del class_dict[ name ]
-                if name[-1:] != '_':
-                    base_traits[ name ] = class_traits[ name ] = value
-                    value_type = value.type
-                    if value_type == 'trait':
-                       handler = value.handler
-                       if handler is not None:
-                           if handler.has_items:
-                               items_trait = _clone_trait(
-                                   handler.items_event(), value.__dict__ )
-
-                               if items_trait.instance_handler == \
-                                   '_list_changed_handler':
-                                   items_trait.instance_handler = \
-                                       '_list_items_changed_handler'
-
-                               class_traits[ name + '_items' ] = items_trait
-
-                           if handler.is_mapped:
-                               class_traits[ name + '_' ] = _mapped_trait_for(
-                                                                         value )
-
-                    elif value_type == 'delegate':
-                        # Only add a listener if the trait.listenable metadata
-                        # is not False:
-                        if value._listenable is not False:
-                            listeners[ name ] = ( 'delegate',
-                                           get_delegate_pattern( name, value ) )
-                    elif value_type == 'event':
-                        on_trait_change = value.on_trait_change
-                        if isinstance( on_trait_change, basestring ):
-                            listeners[ name ] = ( 'event', on_trait_change )
-                else:
-                    name = name[:-1]
-                    prefix_list.append( name )
-                    prefix_traits[ name ] = value
 
-            elif isinstance( value, FunctionType ) or is_cython_func_or_method(value):
-                pattern = getattr( value, 'on_trait_change', None )
-                if pattern is not None:
-                    listeners[ name ] = ( 'method', pattern )
+def create_traits_meta_dict( class_name, bases, class_dict, is_category ):
+    """ Processes all of the traits related data in the class dictionary.
 
-            elif isinstance( value, property ):
-                class_traits[ name ] = generic_trait
+    .. warning:: this function alters the `class_dict` input dictionary.
```

Comment:
Would it be possible to add descriptions of the inputs and return value, using the usual NumPy-like docstring style?